### ISSUE TITLE
[Versions] Asset Versions with S3 Storage not working correctly

### ISSUE THREAD

**Issue by matthiashamacher** [pimcore/pimcore#9609]
# Bug Report
It seems that there is something wrong with the S3 Storage Setup for Versions. For me it seems like either the files cannot be read after creating them in S3 or they are created not correctly.

### Expected behavior
Selecting an Asset Version shows the preview of that version's images.

### Actual behavior
After selecting a version the error "Call to a member function stripImage() on null" is shown. Also getting the version data in PHP returns an empty string.

### Steps to reproduce
1. Configure S3 Storage for `pimcore.version.storage` ([Dokumentation](https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/File_Storage_Setup.html#page_File-Storage-Setup))
2. Upload an asset
3. Update the asset by uploading a new image
4. Select the Versions tab
5. Select a version

**Comment by dvesh3** [comment]
@matthiashamacher sorry, it works for me. Could you please share complete stack trace? seems like this coming from Imagick.

**Comment by matthiashamacher** [comment]
@dvesh3 Just tested it on the newest Version 10.0.7 and still have the problem. The full stack trace is the following:
```
Error:
Call to a member function stripImage() on null

  at vendor/pimcore/pimcore/lib/Image/Adapter/Imagick.php:215
  at Pimcore\Image\Adapter\Imagick->save('/var/www/pimcore/var/tmp/image-version-preview-60e6a23db90e6.png', 'png32')
     (vendor/pimcore/pimcore/lib/Twig/Extension/HelpersExtension.php:104)
  at Pimcore\Twig\Extension\HelpersExtension->getImageVersionPreview('/var/www/pimcore/var/tmp/temp-file-60e6a23db8069-03eeb6eef148c14b5901da5f8eb38e.tmp')
     (/var/application/cache/dev/twig/25/25555e7af48df5c3affa748fadb822e6dcb43cbabbc3c2b8e2425eaf8df28b13.php:77)
  at __TwigTemplate_4f69616923cee35cbeef3b49f577382eb3ded296aec8dfb9cdf91d2aafc792b2->doDisplay(array('asset' => object(Image), 'loader' => object(DataLoader), 'app' => object(AppVariable), 'container' => object(App_KernelDevDebugContainer), 'pimcore_csrf' => object(CsrfProtectionHandler), 'csrfToken' => 'e491235e39fb61d972032bb534a4a0a0e7b3cb7d', 'tempFile' => '/var/www/pimcore/var/tmp/temp-file-60e6a23db8069-03eeb6eef148c14b5901da5f8eb38e.tmp'), array())
     (vendor/twig/twig/src/Template.php:394)
  at Twig\Template->displayWithErrorHandling(array('asset' => object(Image), 'loader' => object(DataLoader), 'app' => object(AppVariable), 'container' => object(App_KernelDevDebugContainer), 'pimcore_csrf' => object(CsrfProtectionHandler), 'csrfToken' => 'e491235e39fb61d972032bb534a4a0a0e7b3cb7d'), array())
     (vendor/twig/twig/src/Template.php:367)
  at Twig\Template->display(array('asset' => object(Image), 'loader' => object(DataLoader)))
     (vendor/twig/twig/src/Template.php:379)
  at Twig\Template->render(array('asset' => object(Image), 'loader' => object(DataLoader)), array())
     (vendor/twig/twig/src/TemplateWrapper.php:40)
  at Twig\TemplateWrapper->render(array('asset' => object(Image), 'loader' => object(DataLoader)))
     (vendor/twig/twig/src/Environment.php:277)
  at Twig\Environment->render('@PimcoreAdmin/Admin/Asset/showVersionImage.html.twig', array('asset' => object(Image), 'loader' => object(DataLoader)))
     (vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php:251)
  at Symfony\Bundle\FrameworkBundle\Controller\AbstractController->renderView('@PimcoreAdmin/Admin/Asset/showVersionImage.html.twig', array('asset' => object(Image), 'loader' => object(DataLoader)))
     (vendor/pimcore/pimcore/lib/Controller/Controller.php:81)
  at Pimcore\Controller\Controller->renderView('@PimcoreAdmin/Admin/Asset/showVersionImage.html.twig', array('asset' => object(Image), 'loader' => object(DataLoader)))
     (vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php:259)
  at Symfony\Bundle\FrameworkBundle\Controller\AbstractController->render('@PimcoreAdmin/Admin/Asset/showVersionImage.html.twig', array('asset' => object(Image), 'loader' => object(DataLoader)), null)
     (vendor/pimcore/pimcore/lib/Controller/Controller.php:43)
  at Pimcore\Controller\Controller->render('@PimcoreAdmin/Admin/Asset/showVersionImage.html.twig', array('asset' => object(Image), 'loader' => object(DataLoader)))
     (vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/Asset/AssetController.php:1065)
  at Pimcore\Bundle\AdminBundle\Controller\Admin\Asset\AssetController->showVersionAction(object(Request))
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:157)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:79)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:199)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:35)  
```

As visible in the stack trace, you are correct, that it is an Issue with Imagick. But only because the file is empty with which Imagik should have generated the preview. I also looked into the `/var/www/pimcore/var/tmp/temp-file-60e6a23db8069-03eeb6eef148c14b5901da5f8eb38e.tmp` File and it is empty too. So the error must be somewhere before when the data is loaded from S3 or already when it is saved.

**Comment by dvesh3** [comment]
https://github.com/pimcore/pimcore/pull/9700 might fix this issue.

**Comment by brusch** [comment]
@matthiashamacher can you please verify whether #9700 fixes the issue or not?

**Comment by BlackbitDevs** [comment]
Partly solved. Now there is the next problem:
```[2021-07-08 15:45:22] request.CRITICAL: Uncaught PHP Exception Exception: "Call to undefined method getDimensions in class Pimcore\Model\Asset" at <Pimcore-Root>/vendor/pimcore/pimcore/lib/Model/AbstractModel.php line 257```
Our customer uses Google Cloud Storage but I think the problem is the same. 

Ah, this only happens for old versions. I just resaved the asset and now it works.
Reason was that the old version is of class `Pimcore\Model\Asset` and not `Pimcore\Model\Asset\Image` but it has `type=image`. For this reason the image template gets loaded in 
https://github.com/pimcore/pimcore/blob/43366c52dbd7c1375b0dbdbeaa43c34d4c2409e2/bundles/AdminBundle/Controller/Admin/Asset/AssetController.php#L1071-L1077
But of course `Pimcore\Model\Asset` does not have a `getDimensions` method.
But it is solved for me as this was only a consequence of the fixed bug #9700

**Comment by matthiashamacher** [comment]
I can confirm that #9700 fixes the issue also for S3.
